### PR TITLE
chore: turn scripts & gulpfile into typescript 

### PR DESCRIPTION
**Pull request by ghost**
#### Pre-flight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

- turn script legacy file into typescript 
- turn gulpfile babel file into typescript
- add type check command in package

#### Motivation and Context

- [ Convert to typescript #42 ](https://github.com/ferdium/ferdium-app/issues/42)

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,9 +26,9 @@
     "dev": "cross-env NODE_ENV=development gulp dev",
     "test": "jest",
     "test:watch": "jest --watch",
-    "lint": "npx eslint \"{src,test,scripts}/**/*.{js,jsx,ts,tsx}\"",
-    "lint:fix": "npx eslint --fix \"{src,test,scripts}/**/*.{js,jsx,ts,tsx}\"",
-    "extract": "formatjs extract \"src/**/*.{js,jsx,ts,tsx}\" --out-file temp.json --flatten --id-interpolation-pattern '[sha512:contenthash:base64:6]' --preserve-whitespace",
+    "lint": "tsc --noEmit && eslint \"{src,test,scripts}/**/*.{js,jsx,ts,tsx}\"",
```
Comment (vraravam) [inline package.json:29]: Isn't this `tsc --noEmit` also required for the next line?

File: gulpfile.babel.ts [hunk 2]
```diff
@@ -185,78 +162,72 @@ export function styles() {
...
 }
 
-export function recipes() {
+function recipes() {
   return gulp
     .src(paths.recipes.src, { since: gulp.lastRun(recipes) })
     .pipe(gulp.dest(paths.recipes.dest));
 }
 
-export function recipeInfo() {
```
Comment (vraravam) [inline gulpfile.babel.ts:237]: Removing all the other `export` - is that due to the reason that they are not invoked from other files anyways?

**Review by vraravam — CHANGES_REQUESTED** [review]

File: gulpfile.babel.ts (hunk 2 above)
Comment (ghost) [reply gulpfile.babel.ts:237]: yes, kind of. As far as i have checked, the gulp file should dev and build variable to be exported.
I have used `CLEAN=true ./scripts/build-unix.sh` to build it and it worked in local.

File: package.json (hunk 1 above)
Comment (ghost) [reply package.json:29]: No, the <b>error</b> `tsc --noEmit` brings need to be manually addressed

<b>error</b> - type error as per tsconfig

**Review by vraravam — APPROVED** [review]

**Comment by vraravam** [comment]
Is this ready for review @muhamedsalih-tw ?

**Comment by ghost** [comment]
> Is this ready for review @muhamedsalih-tw ?

yes, I shall mark it as ready for review

**Comment by vraravam** [comment]
@all-contributors please add @muhamedsalih-tw for code

**Pull request merged**